src/weather_model_graphs/create/mesh/mesh.py

```diff
@@ -46,10 +46,13 @@ def create_single_level_2d_mesh_graph(xy, nx, ny):
     ly = np.linspace(ym + dy / 2, yM - dy / 2, ny)
 
     mg = np.meshgrid(lx, ly)
-    g = networkx.grid_2d_graph(len(ly), len(lx))
+    g = networkx.grid_2d_graph(len(lx), len(ly))
```

Comment:
I'm used to matrix notation where MxN means M rows and N columns. Also the arguments to grid_2d_graph are (m, n). So I'd rather see xy [2,M,N] where M and N are grid points in the y and x direction respectively and g = networkx.grid_2d_graph(len(ly), len(lx))

Reply:
Hmm, yes this part is a little bit confusing. I think the problem is that this determines the node name and what goes into the `pos` attributes. While I agree that `xy` is best viewed as having shape `[2,M,N]`, I think it also makes sense that node positions/coordinates are of the form `(x,y)` rather than `(y,x)`. This passage of the code is what determines this, by the order of arguments passed to `grid_2d_graph`.

Below this section you have the code 
```python
# add diagonal edges
    g.add_edges_from(
        [((x, y), (x + 1, y + 1)) for x in range(nx - 1) for y in range(ny - 1)]
        + [((x + 1, y), (x, y + 1)) for x in range(nx - 1) for y in range(ny - 1)]
    )
```
which I think makes sense, and again makes the assumption of `(x,y)` positions. 

What I did now is to clarify the docstring of this function to correctly match M to y and N to x, and  add a comment to make it explicit that node names and positions are `(x,y)` from here on.